### PR TITLE
replace `"image"` and `"label"` with `CommonKeys.IMAGE` and `CommonKeys.LABEL`

### PR DESCRIPTION
**Pull request by rijobro**
Fixes https://github.com/Project-MONAI/MONAI/issues/3603.

This prepares the codebase for tracking meta data.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by wyli — COMMENTED** [review]
I feel `DictPostFixes.META` is quite useful for linking data and metadata. not sure about `CommonKeys.IMAGE` and `CommonKeys.LABEL`, the "image" is more intuitive, and we shouldn't rely on these for data typing, as it will introduce framework-like contraints... cc @ericspod @Nic-Ma

**Comment by rijobro** [comment]
Externally, users can still use `"image"`, I just thought that internally we should avoid using "magic" variables where possible which means converting `"image"` to `CommonKeys.IMAGE` etc. 

I can revert this component of the PR if you would like.

**Comment by ericspod** [comment]
In general we shouldn't have magic strings to avoid issues with typos or using inconsistent names in places, our code should exhibit what we believe to be good practice. Users can indeed still use string literals if they want so they aren't constrained to using our constants. It is more verbose this way but I don't feel it's unintuitive, the names of our constants should be as descriptive as the string literals they replace.

**Comment by wyli** [comment]
> the names of our constants should be as descriptive as the string literals they replace.

So how about renaming `CommonKeys.Image` to `DefaultKeyStr.Image`? or how to make sure we don't give an impression that the keys must be some object/class defined by MONAI? The keys can be any Hashable...

also, would be great to have a separate PR to introduce `DictPostFixes`, looks like we all happy with it.

**Comment by rijobro** [comment]
Ok, this PR has been updated to only include `CommonKeys` changes.

File: tests/test_deepgrow_transforms.py [hunk 1]
```diff
@@ -73,42 +79,42 @@
...
+    CommonKeys.IMAGE: np.arange(1000).reshape((1, 5, 10, 20)),
+    f"{CommonKeys.IMAGE}_meta_dict": {"foreground_cropped_shape": (1, 10, 20, 40)},
     "guidance": [[[6, 10, 14], [8, 10, 14]], [[8, 10, 16]]],
 }
 
 DATA_10 = {
-    "image": np.arange(9).reshape((1, 1, 3, 3)),
-    "image_meta_dict": {
+    CommonKeys.IMAGE: np.arange(9).reshape((1, 1, 3, 3)),
+    f"{CommonKeys.IMAGE}_meta_dict": {
```
Comment (ericspod) [inline tests/test_deepgrow_transforms.py:117]: The "_meta_dict" part is now like a magic string, should we have that stored in `CommonKeys` and have `f"{CommonKeys.Image}_{CommonKeys.MetaDict}"`?

**Comment by ericspod** [comment]
I tried going through everything but there's a lot of changes but it looks good what I saw. The only thing I mentioned was using "_meta_dict" as a literal in places. If we instead had `CommonKeys.MetaDict` then we could make keys like `f"{CommonKeys.Image}_{CommonKeys.MetaDict}"` or access dictionaries like `d[CommonKeys.Image, CommonKeys.MetaDict] = x` which will associate `x` with a pair containing the two key values. If we went with this approach there would be a lot more changes to make to use that syntax so not in this PR.

**Review by ericspod — APPROVED** [review]

File: tests/test_deepgrow_transforms.py (hunk 1 above)
Comment (rijobro) [reply tests/test_deepgrow_transforms.py:117]: see https://github.com/Project-MONAI/MONAI/pull/3671.

Comment (ericspod) [reply tests/test_deepgrow_transforms.py:117]: Yeah that looks good though we should commit this PR first.

**Pull request closed without merging**